python/lsst/ts/genericcamera/genericcameracsc.py

```diff
@@ -1005,6 +1000,11 @@ async def configure(self, config):
                 f"No config found for sal_index={self.salinfo.index}"
             )
 
+        if config.s3instance != "none":
+            self.use_lfa = True
+        if config.s3instance == "mock":
```

Comment:
I think the same I mentioned above also applies here. If you configure the CSC with `config.s3instance == "mock"` any subsequent pass will continue to leave `self.s3_mock = True`. 

Consider doing:

```
self.s3_mock = config.s3instance == "mock"
```

Reply:
Implemented.